### PR TITLE
Porting Gemma 2 transformers checkpoint

### PR DESCRIPTION
**Pull request by ariG23498**
Porting Gemma 2 transformers checkpoints in Keras NLP

### CODE REVIEW

**Review by mattdangerw — COMMENTED** [review]
Thanks! This all looks good to me. Let's add a test.

**Comment by ariG23498** [comment]
@mattdangerw @grasskin this PR is ready for review!

Note: The KerasNLP Gemma 2 model works only on the JAX backend (for the time being)

File: keras_nlp/src/utils/transformers/convert_gemma.py [hunk 1]
```diff
@@ -77,10 +103,29 @@ def load_gemma_backbone(cls, preset, load_weights):
...
+        if transformers_config["model_type"] == "gemma":
+            port_weight(
+                keras_variable=decoder_layer.pre_ffw_norm.variables[0],
+                hf_weight_key=f"model.layers.{i}.post_attention_layernorm.weight",
+            )
+        elif transformers_config["model_type"] == "gemma2":
+            port_weight(
+                keras_variable=decoder_layer.pre_ffw_norm.variables[0],
+                hf_weight_key=f"model.layers.{i}.pre_feedforward_layernorm.weight",
+            )
```
Comment (ariG23498) [inline keras_nlp/src/utils/transformers/convert_gemma.py:122]: This was done in order to align the gemma 1 and gemma 2 checkpoints.

I am open to better ways to go around it.

**Comment by mattdangerw** [comment]
Thanks!

**Pull request merged**